src/codegen/ast.lisp

```diff
@@ -112,8 +113,19 @@
 ;;;
 
 
-(defstruct (node (:constructor nil))
-  (type (util:required 'type) :type tc:ty :read-only t))
+(defstruct (node (:constructor nil)
```

Comment:
this is going to export (setf node-type). My recommendation is to avoid doing that, personally, as it's not allowed by the API. But it means we'll need the %type shenanigans, or a different conc-name.

Reply:
Yeah, I ended up switching it back from `%type` to `type` because there are like 1000 places in the code where node structs are created using `:type`. Do you know a nice way of handling that? I tried a few things and ended up writing another stupid 18-case generic function to allow `copy-node` with editing some fields, without changing any of the prior AST node struct definitions.

Reply:
Yes, add

    (:conc-name %node-)

just before the `(:constructor nil)` clause so that the accessor becomes

    %node-type

and

    (setf %node-type)

Define your own

    node-type

reader based off of this. The `:node` keyword argument will still work.